### PR TITLE
cloudflared: Fix incorrect uci config syntax

### PR DESCRIPTION
**Pull request by Ra2-IFV**
Maintainer: @1715173329  
Compile tested: aarch64 a24c3d  
Run tested: aarch64 a24c3d  
Description:  
`append_param_arg()` modified  
need discussion to approve adding more parameters

### CODE REVIEW

File: net/cloudflared/files/cloudflared.config [hunk 1]
```diff
@@ -4,10 +4,14 @@ config cloudflared 'config'
 	option token ''
 	option config '/etc/cloudflared/config.yml'
 	option origincert '/etc/cloudflared/cert.pem'
-	option edge-bind-address 'auto'
-	option protocol 'auto'
+	option edge_bind_address ''
+	option edge_ip_version ''
+	option grace_period ''
+	option protocol ''
```
Comment (1715173329) [inline net/cloudflared/files/cloudflared.config:10]: should be `http2` for now

File: net/cloudflared/files/cloudflared.init [hunk 2]
```diff
@@ -9,8 +9,17 @@ PROG="/usr/bin/cloudflared"
 
 append_param_arg() {
 	local value
+	# Use $1 as key, get value from uci
+	# If value empty, append $1 as param, $2 as value
```
Comment (1715173329) [inline net/cloudflared/files/cloudflared.init:13]: these comments are really unnecessary

File: net/cloudflared/files/cloudflared.init [hunk 3]
```diff
@@ -9,8 +9,17 @@ PROG="/usr/bin/cloudflared"
...
 	config_get value "config" "$1" $2
-	[ -n "$value" ] && procd_append_param command "--$1" "$value"
+	# If value not empty, value as value
+	[ -n "$value" ] && if [ -n "$3" ]; then
+		# If has $3, append $3 as param
+		procd_append_param command "--$3" "$value"
```
Comment (1715173329) [inline net/cloudflared/files/cloudflared.init:18]:
```suggestion
	[ -n "$value" ] && procd_append_param command "--${1//_/-}" "$value"
```

keep it simple

**Review by 1715173329 — CHANGES_REQUESTED** [review]

**Comment by 1715173329** [comment]
and missing PKG_RELEASE bump.

File: net/cloudflared/files/cloudflared.config (hunk 1 above)
Comment (Ra2-IFV) [reply net/cloudflared/files/cloudflared.config:10]: Should we really set `http2` by default? At least for me the default behavior, which uses `quic` first, fallback to `http2` if failed, works well, and most users should benefit from this. Some users may need to set `http2` explicitly as their ISP is not quic-friendly, reported in this [forum post](https://community.cloudflare.com/t/cloudflared-keeps-falling-back-from-quic-to-http2/385031).

File: net/cloudflared/files/cloudflared.init (hunk 2 above)
Comment (Ra2-IFV) [reply net/cloudflared/files/cloudflared.init:13]: Agree :)

File: net/cloudflared/files/cloudflared.init (hunk 3 above)
Comment (Ra2-IFV) [reply net/cloudflared/files/cloudflared.init:18]: Agree, I don't know what writing style should I use so I chose `readability` for default.

**Comment by Ra2-IFV** [comment]
> and missing PKG_RELEASE bump.

Fixed. One more mistake and one more important thing to remind myself

File: net/cloudflared/files/cloudflared.config (hunk 1 above)
Comment (Ra2-IFV) [reply net/cloudflared/files/cloudflared.config:10]: Github reviewing is nice...

Comment (1715173329) [reply net/cloudflared/files/cloudflared.config:10]: Nah, `quic` requires cloudflare patched go toolchain, which is not supported here.
Only `http2` is functional currently, so this is the reason that PR was created.

Comment (Ra2-IFV) [reply net/cloudflared/files/cloudflared.config:10]: Copy sir

File: net/cloudflared/files/cloudflared.init [hunk 4]
```diff
@@ -10,7 +10,7 @@ PROG="/usr/bin/cloudflared"
 append_param_arg() {
 	local value
 	config_get value "config" "$1" $2
-	[ -n "$value" ] && procd_append_param command "--$1" "$value"
+  [ -n "$value" ] && procd_append_param command "--${1//_/-}" "$value"
```
Comment (1715173329) [inline net/cloudflared/files/cloudflared.init:13]: use tab instead

File: net/cloudflared/files/cloudflared.init [hunk 5]
```diff
@@ -26,10 +26,14 @@ start_service() {
 
 	append_param_arg "config" "/etc/cloudflared/config.yml"
 	append_param_arg "origincert" "/etc/cloudflared/cert.pem"
-	append_param_arg "edge-bind-address"
+	append_param_arg "edge_bind_address" "" "edge-bind-address"
```
Comment (1715173329) [inline net/cloudflared/files/cloudflared.init:29]:
```suggestion
	append_param_arg "edge_bind_address"
```

the same below

**Review by 1715173329 — APPROVED** [review]

File: net/cloudflared/files/cloudflared.init (hunk 3 above)
Comment (Ra2-IFV) [reply net/cloudflared/files/cloudflared.init]: Oh, it can written in that way!  
Learned this, thanks!

**Comment by moetayuko** [comment]
@1715173329 As indicated by https://github.com/cloudflare/cloudflared/commit/43a3ba347b49d75ae6b92d5e040ed5a17a616ca6, the post quantum curves are already available in go 1.24. I guess it's time to revert protocol 'http2' back to 'auto'

**Comment by 1715173329** [comment]
> I guess it's time to revert protocol 'http2' back to 'auto'

#26945

**Pull request merged**